bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2BridgeHandler.java

```diff
@@ -0,0 +1,774 @@
+                properties.put(PROPERTY_RESOURCE_TYPE, device.getType().toString());
+
+                // set metadata properties
+                String metaDataName = metaData.getName();
+                if (Objects.nonNull(metaDataName)) {
+                    properties.put(PROPERTY_RESOURCE_NAME, metaDataName);
+                }
+                properties.put(PROPERTY_RESOURCE_ARCHETYPE, metaData.getArchetype().toString());
+
+                // set product data properties
+                ProductData productData = device.getProductData();
+                if (Objects.nonNull(productData)) {
+                    // set generic thing properties
+                    properties.put(Thing.PROPERTY_MODEL_ID, productData.getModelId());
+                    properties.put(Thing.PROPERTY_VENDOR, productData.getManufacturerName());
+                    properties.put(Thing.PROPERTY_FIRMWARE_VERSION, productData.getSoftwareVersion());
+                    String hardwarePlatformType = productData.getHardwarePlatformType();
+                    if (Objects.nonNull(hardwarePlatformType)) {
+                        properties.put(Thing.PROPERTY_HARDWARE_VERSION, hardwarePlatformType);
+                    }
+
+                    // set hue specific properties
+                    properties.put(PROPERTY_PRODUCT_NAME, productData.getProductName());
+                    properties.put(PROPERTY_PRODUCT_ARCHETYPE, productData.getProductArchetype().toString());
+                    properties.put(PROPERTY_PRODUCT_CERTIFIED, productData.getCertified().toString());
+                }
+                break; // we only needed the BRIDGE_V2 resource
+            }
+        }
+        thing.setProperties(properties);
+    }
+
+    /**
+     * Update the thing's own state. Called sporadically in case any SSE events may have been lost.
+     */
+    private void updateSelf() {
+        logger.debug("updateSelf()");
+        try {
+            checkAssetsLoaded();
+            updateProperties();
+            getClip2Bridge().open();
+        } catch (ApiException e) {
+            logger.trace("updateSelf() {}", e.getMessage(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/offline.api2.comm-error.exception [\"" + e.getMessage() + "\"]");
+            onConnectionOffline();
+        } catch (AssetNotLoadedException e) {
+            logger.trace("updateSelf() {}", e.getMessage(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.api2.conf-error-assets-not-loaded");
+        } catch (HttpUnauthorizedException e) {
+            logger.trace("updateSelf() {}", e.getMessage(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.api2.conf-error-access_denied [\"" + e.getMessage() + "\"]");
```

Comment:
This cannot happen. I'm not sure if intended or by mistake, but it seems in all cases to be internally caught and rethrown as `ApiException`.

Reply:
Good catch. The intention is that `testConnectionState()` can throw both HttpUnauthorizedException and APIException. Whereas HttpUnauthorizedException is converted to ApiException for all other calls.

Reply:
Fixed.